src/Arcus.Templates.AzureFunctions.ServiceBus.Queue/Startup.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Arcus.Security.Core.Caching.Configuration;
+using Arcus.Templates.AzureFunctions.ServiceBus.Queue;
+using Arcus.Templates.AzureFunctions.ServiceBus.Queue.Model;
+using Microsoft.Azure.Functions.Extensions.DependencyInjection;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Serilog;
+using Serilog.Configuration;
+using Serilog.Events;
+
+[assembly: FunctionsStartup(typeof(Startup))]
+
+namespace Arcus.Templates.AzureFunctions.ServiceBus.Queue
+{
+    public class Startup : FunctionsStartup
+    {
+        // This method gets called by the runtime. Use this method to configure the app configuration.
+        public override void ConfigureAppConfiguration(IFunctionsConfigurationBuilder builder)
+        {
+            builder.ConfigurationBuilder.AddEnvironmentVariables();
+        }
+
+        /// <summary>
+        /// This method gets called by the runtime. Use this method to add services to the container.
+        /// </summary>
+        /// <param name="builder">The instance to build the registered services inside the functions app.</param>
+        public override void Configure(IFunctionsHostBuilder builder)
+        {
+            builder.ConfigureSecretStore((context, config, stores) =>
+            { 
+//[#if DEBUG]
+                stores.AddConfiguration(config);
+//[#endif]
+
+                //#error Please provide a valid secret provider, for example Azure Key Vault: https://security.arcus-azure.net/features/secret-store/provider/key-vault
+                stores.AddAzureKeyVaultWithManagedIdentity("https://your-keyvault.vault.azure.net/", CacheConfiguration.Default);
+            });
+
+            builder.AddServiceBusMessageRouting()
+                   .WithServiceBusMessageHandler<OrdersAzureServiceBusMessageHandler, Order>();
+
+            LoggerConfiguration logConfig = CreateLoggerConfiguration(builder);
+            builder.Services.AddLogging(logging =>
+            {
+                logging.AddSerilog(logConfig.CreateLogger(), dispose: true);
+            });
+        }
+
+        private static LoggerConfiguration CreateLoggerConfiguration(IFunctionsHostBuilder builder)
+        {
+            IConfiguration appConfig = builder.GetContext().Configuration;
+            var instrumentationKey = appConfig.GetValue<string>("APPINSIGHTS_INSTRUMENTATIONKEY");
+            var logConfig = new LoggerConfiguration()
+                .MinimumLevel.Debug()
+                .MinimumLevel.Override("Microsoft", LogEventLevel.Information)
+                .Enrich.FromLogContext()
+                .Enrich.WithComponentName("Service Bus Queue Trigger")
+                .Enrich.WithVersion()
+                .WriteTo.Console()
+                .WriteTo.AzureApplicationInsights(instrumentationKey);
```

Comment:
I always tend to only add the Application Insights sink if the instrumentation-key is provided. (not empty).
It's debatable :)

Reply:
Yes, I saw that of you 😉 . I see if the other templates have this as well, if so, I'm happy to add it here. 👍

Reply:
Hmm, I can't find it in other templates, but I wouldn't mind if we do add this to all our templates. I do, however would like this to be logged when the instrumentation key is blank, because it could lead to unexpected results if we 'magically' skip the Azure Application Insights Serilog sink from being added.

Reply:
What is the benefit from checking first the instrumentation key? That the project don't gets stopped because of a missing logging configuration?

Reply:
If you do not have this setting configured, the project will crash and fail to start.
You can look at it in 2 ways:
- either you leave it as is, and your program crashes.   Then let the developer decide if he wants to remove the sink to Application Insights, or needs to add the required setting
- Only add the application insights sink if the instrumention-key is provided.  If it is not provided, log a warning.   I've used this approach in one of my projects because I had situations where my application needed to log to App Insights, but when deployed on prem (on a vessel), no app-insights logging should happen.

Reply:
Ok, great! Then let's add this to all our templates and make sure our user-experience is even better! 😉 
https://github.com/arcus-azure/arcus.templates/issues/563